src/snowflake/snowpark/functions.py

```diff
@@ -9830,16 +9830,46 @@ def make_interval(
     You can also find some examples to use interval constants with :meth:`~snowflake.snowpark.Window.range_between`
     method.
     """
-
+    ast = None
     if _emit_ast:
-        raise NotImplementedError("TODO SNOW-1690923: Add interval support to IR.")
+        ast = proto.Expr()
+        # Encode the parameters as kwargs to make them more readable.
+        # If any of the parameters are None, ignore them.
+        kwargs = {}
+        if years:
```

Comment:
use `is not None`. Technically, `years=0` will also not be encoded.

Reply:
fixed!